pkg/new-ui/v1beta1/backend.go

```diff
@@ -421,3 +431,92 @@ func (k *KatibUIHandler) FetchTrial(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+// FetchTrialLogs fetches logs for a trial in specific namespace.
+func (k *KatibUIHandler) FetchTrialLogs(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Requesting logs")
+
+	trialName := r.URL.Query()["trialName"][0]
+	namespace := r.URL.Query()["namespace"][0]
+	log.Printf("Requesting logs")
+
+	logs, err := getTrialLogs(k, trialName, namespace)
+	if err != nil {
+		log.Printf("GetLogs failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	response, err := json.Marshal(logs)
+	if err != nil {
+		log.Printf("Marshal logs failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if _, err = w.Write(response); err != nil {
+		log.Printf("Write logs failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+// GetTrialLogs returns logs of a master Pod for the given job name and namespace
+func getTrialLogs(k *KatibUIHandler, trialName string, namespace string) (string, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return "", err
+	}
+
+	clientset, err := corev1.NewForConfig(cfg)
+	if err != nil {
+		return "", err
+	}
+
+	trial := &trialsv1beta1.Trial{}
+	if err := k.katibClient.GetClient().Get(context.TODO(), types.NamespacedName{Name: trialName, Namespace: namespace}, trial); err != nil {
+		return "", err
+	}
```

Comment:
Should we verify the auth here to check if user can get the Pod logs, similar to [other API calls](https://github.com/kubeflow/katib/blob/master/pkg/new-ui/v1beta1/backend.go#L108) ?
cc @apo-ger @kimwnasptd

Reply:
@andreyvelich I think we should. This is the required authorization check before fetching the trial:

```go
user, err := IsAuthorized(consts.ActionTypeGet, namespace, consts.PluralTrial, "", trialName, trialv1beta1.SchemeGroupVersion, k.katibClient.GetClient(), r)
if user == "" && err != nil {
	log.Printf("No user provided in kubeflow-userid header.")
	http.Error(w, err.Error(), http.StatusUnauthorized)
	return
} else if err != nil {
	log.Printf("The user: %s is not authorized to delete tial: %s in namespace: %s \n", user, trialName, namespace)
	http.Error(w, err.Error(), http.StatusForbidden)
	return
}

```
Similar to what we do in [new-ui/v1beta1/hp.go](https://github.com/kubeflow/katib/blob/bd91301d36051d4cf04fe88bb3751591df4141b9/pkg/new-ui/v1beta1/hp.go#L98-L107)

Reply:
Thanks @apo-ger for the code snippet!